### ISSUE TITLE
Refactoring of the CircularDependencyDetector::registerDependencyWait

### ISSUE THREAD

**Issue by ckampm** [ChimeraTK/ApplicationCore#202]
Note : This ticket should be done after #201.

A possible deadlock was resolved in 8f99d74. However, access to the members is spread across the function which makes it hard to reason about the locking. Check the code and rearrange member access so that the critical sections become more compact and prominent.

**Comment by mhier** [comment]
superseeded by redmine #8011 (which is already done)
